### ISSUE TITLE
memory leak

### ISSUE THREAD

**Issue by BalticPinguin** [libMesh/libmesh#1982]
Dear all,
I have run my application with valgrind, which revealed some memory-leak.
Unfortunately, I am not very familiar with valgrind but for me, this looks like an error related to libmesh.
I also don't understand, why only few of the 21 records are listed; the attached file contains the complete valgrind-output.

I'd appreciate any help in this respect. Probably I am handling the unique_ptr wrong?

[memory_leak.txt](https://github.com/libMesh/libmesh/files/2716268/memory_leak.txt)

Best regards and a good start in the new year to all,
Hubert

**Comment by roystgnr** [comment]
In theory the InfFE::base_qrule heap allocation ought to be freed when the InfFE is destroyed, to look at the inf_fe.C:81 example.  How is that InfFE being allocated and freed?

**Comment by BalticPinguin** [comment]
Thanks for the hints; I have fixed it now:
The problem was that I replaced the object where an auto_ptr pointet at using ``qrule.reset(new QGauss(...))``, which seemingly whereas``qrule.release();`` and than setting qrule to a newly build object works.
Even though I would understand the [documentation](http://www.cplusplus.com/reference/memory/auto_ptr/reset/) such that it should work.
Unfortunately, the leak points to the creation of the first object, so I didn't find it.
Anyway: Thank you for that hints.
